src/main/java/com/kamycz3q/coursesystemspringboot/personalData/controllers/PersonalDataController.java

```diff
@@ -0,0 +1,46 @@
+package com.kamycz3q.coursesystemspringboot.personalData.controllers;
+
+
+import com.kamycz3q.coursesystemspringboot.personalData.personalDataObject.PersonalData;
+import com.kamycz3q.coursesystemspringboot.personalData.record.CreatePersonalRequest;
+import com.kamycz3q.coursesystemspringboot.personalData.service.PersonalDataService;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/personal-data/personal")
+public class PersonalDataController {
+    private final PersonalDataService personalDataService;
+
+    @Autowired
+    public PersonalDataController(PersonalDataService personalDataService) {
+        this.personalDataService = personalDataService;
+    }
+
+    @GetMapping()
+    public List<PersonalData> listPersonalData() {
+        return personalDataService.getPersonalDataList();
+    }
+
+    @GetMapping("/{id}")
+    public PersonalData getPersonalData(@PathVariable("id") String id) {
+        return personalDataService.getPersonalData(Long.parseLong(id));
+    }
+
+    @PostMapping("")
+    public PersonalData addPersonalData(@NotNull @RequestBody CreatePersonalRequest createPersonalRequest) {
```

Comment:
Bardzo dobrze, że zrobiłeś dedykowaną klasę na parametry przychodzące w requeście. Niedobrze, że zwracasz w endpointcie klasę `@Entity`. W ten sposób Twój model bazodanowy przecieka do API. To co serwujemy klientowi, zawsze powinno być dedykowaną strukturką danych. Taka klasa, która służy jedynie do transportu określonych danych, nazywa się DTO (data transfer object). Jej sensem istnienia jest tylko bycie przekazaną skądś dokądś i zwykle pomiędzy jakimiś znaczącymi warstwami aplikacji lub jest ona wpuszczana na wejściu do aplikacji i na wyjściu. Twoja klasa `CreatePersonalRequest` to też tak na prawdę DTO. Encja bazodanowa (`@Entity`) też jest w jakimś ujęciu klasą DTO, bo służy do transportu danych do bazy. Encja jest natomiast wyspecjalizowana i robi dużo więcej, bo zawiera też mapowania np. do konkretnych kolumn i tabel w bazie.

Te adnotacje @NotNull to nie powinny być z pakietu .jetbrains, tylko ze springa.

Reply:
adnotacje i zwracane wartosci zmienione na DTO